hooks/terraform_providers_lock.sh

```diff
@@ -19,6 +19,59 @@ function main {
   common::per_dir_hook "$HOOK_ID" "${#ARGS[@]}" "${ARGS[@]}" "${FILES[@]}"
 }
 
+#######################################################################
+# Check that all needed `h1` and `zh` SHAs are included in lockfile for
+# each provider.
+# Arguments:
+#   platforms_count (number) How many `-platform` flags provided
+# Outputs:
+#   Return 0 when lockfile has all needed SHAs
+#   Return 1-99 when lockfile is invalid
+#   Return 100+ when not all SHAs found
+#######################################################################
+function lockfile_contains_all_needed_sha {
+  local -r platforms_count="$1"
+
+  local h1_counter="$platforms_count"
+  local zh_counter=0
+
+  # Reading each line
+  while read -r line; do
+
+    if [ "$(echo "$line" | grep -o '^"h1:')" == '"h1:' ]; then
+      h1_counter=$((h1_counter - 1))
+      continue
+    fi
+
+    if [ "$(echo "$line" | grep -o '^"zh:')" == '"zh:' ]; then
+      zh_counter=0
+      continue
+    fi
+
+    if [ "$(echo "$line" | grep -o ^provider)" == "provider" ]; then
+      h1_counter="$platforms_count"
+      zh_counter=$((zh_counter + 1))
+      continue
+    fi
+    # No all SHA in provider found
+    if [ "$(echo "$line" | grep -o '^}')" == "}" ]; then
+      if [ "$h1_counter" -ge 1 ] || [ "$zh_counter" -ge 1 ]; then
+        return $((150 + h1_counter + zh_counter))
```

Comment:
The intro to this function declared 
"100+" rather than "150+"

Reply:
100+ included 150+ :) 

Reason:

>The code also relies on the assertion that number of hashes equals to number of -platform flags passed to terraform providers lock, which — based on my experience — in some cases is not true.

Reply:
h1_counter can be `-1`

Reply:
Is this comment would be sufficient?
```suggestion
        # h1_counter can be less than 0, in the case when lockfile
        # contains more platforms than you currently specify
        # That's why here extra +50 - for safety reasons, to be sure
        # that error goes exactly from this part of the function
        return $((150 + h1_counter + zh_counter))
```

Reply:
Yep, it is. Though I still am not sure why need to return dynamic values for only three types of results 🤷🏻

Reply:
For debug proposes